src/ipc/handler.c

```diff
@@ -74,6 +74,7 @@ static inline struct sof_ipc_hdr *mailbox_validate(void)
 {
 	struct sof_ipc_hdr *hdr = _ipc->comp_data;
 
+#if !defined CONFIG_SUECREEK
 	/* read component values from the inbox */
 	mailbox_hostbox_read(hdr, 0, sizeof(*hdr));
```

Comment:
ok, so lets make mailbox_read on SUE read the SPI buffer, that way we wont need the #ifdef